### PR TITLE
vrrp: T6020: vrrp health-check script not applied correctly

### PR DESCRIPTION
**Pull request by HollyGurza**
T6020: vrrp health-check script not applied correctly in keepalived.conf

Added health check to sync-group in CLI
Don't use instance health-check when instance in sync group member
Warning notify for health-check config problem
New smoke test

## Change Summary

Added health check to sync-group in CLI
Don't use instance health-check when instance in sync group member
Warning notify for health-check config problem
New smoke test
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T6020

## Proposed changes

high-availability vrrp

## Smoketest result

```
vyos@vyos# python3 /usr/libexec/vyos/tests/smoke/cli/test_high-availability_vrrp.py 
test_01_default_values (__main__.TestVRRP.test_01_default_values) ... ok
test_02_simple_options (__main__.TestVRRP.test_02_simple_options) ... ok
test_03_sync_group (__main__.TestVRRP.test_03_sync_group) ... ok
test_04_exclude_vrrp_interface (__main__.TestVRRP.test_04_exclude_vrrp_interface) ... ok
test_check_health_script (__main__.TestVRRP.test_check_health_script) ... ok

----------------------------------------------------------------------
Ran 5 tests in 43.278s

OK

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

**Review by zdc — CHANGES_REQUESTED** [review]
In the history of this template, I see that this logic was changed several times, many of them with mistakes.

Would it not be better to finally add health-check to sync-group in CLI and disallow configurations with health-check inside the vrrp group if it is a group member?

File: data/templates/high-availability/keepalived.conf.j2 [hunk 1]
```diff
@@ -128,7 +128,7 @@ vrrp_instance {{ name }} {
 {%             endfor %}
     }
 {%         endif %}
-{%         if group_config.health_check is vyos_defined %}
+{%         if group_config.health_check is vyos_defined and group_config.health_check.script is not vyos_defined %}
```
Comment (c-po) [inline data/templates/high-availability/keepalived.conf.j2]: `{%         if group_config.health_check.script is not vyos_defined %}` would be sufficient in this case

**Comment by faekz0r** [comment]
> In the history of this template, I see that this logic was changed several times, many of them with mistakes.
> 
> Would it not be better to finally add health-check to sync-group in CLI and disallow configurations with health-check inside the vrrp group if it is a group member?

That sounds like a great idea. In fact when I ran into the current bug, I was astounded that you can't actually configure the health-check for the vrrp group from the VyOS CLI.
Otherwise I could have also worked around the bug by configuring it correctly from CLI.

**Comment by HollyGurza** [comment]
> In the history of this template, I see that this logic was changed several times, many of them with mistakes.
> 
> Would it not be better to finally add health-check to sync-group in CLI and disallow configurations with health-check inside the vrrp group if it is a group member?

Done

**Comment by sever-sever** [comment]
@zdc Could you check the proposed changes?

**Review by dmbaturin — APPROVED** [review]
I don't see any issues with the implementation.
@zdc could you check if everything you wanted to address is done?

**Comment by dmbaturin** [comment]
@mergify backport sagitta

**Comment by dmbaturin** [comment]
@mergifyio backport sagitta

**Comment by sever-sever** [comment]
@MergifyIo backport sagitta

**Comment by sever-sever** [comment]
@HollyGurza Could you create PR for 1.4? 
For some reason, mergify does not work

**Comment by HollyGurza** [comment]
> @HollyGurza Could you create PR for 1.4? For some reason, mergify does not work

https://github.com/vyos/vyos-1x/pull/3122

**Pull request merged**